jmh-benchmarks/src/main/java/org/apache/kafka/jmh/core/TestPurgatoryPerformance.java

```diff
@@ -479,7 +479,8 @@ public void onComplete() {
 
         @Override
         public boolean tryComplete() {
-            return System.currentTimeMillis() >= completesAt && forceComplete();
+            forceComplete();
+            return System.currentTimeMillis() >= completesAt && isCompleted();
```

Comment:
Shouldn't this be as below?
```suggestion
            if (System.currentTimeMillis() >= completesAt) {
                forceComplete();
                return true;
            }
            return false;
```